#!/bin/sh /usr/share/dpatch/dpatch-run
## 16-fix-random-actors.dpatch by Michael Karcher
## <debian@mkarcher.dialup.fu-berlin.de>
##
## DP: Fix direction of random opponents.

@DPATCH@
diff -urNad rott-1.0+dfsg~/rt_ted.c rott-1.0+dfsg/rt_ted.c
--- rott-1.0+dfsg~/rt_ted.c	2008-01-10 09:18:47.000000000 +0100
+++ rott-1.0+dfsg/rt_ted.c	2008-01-10 09:18:48.000000000 +0100
@@ -5897,6 +5897,7 @@
  int starti,totalrandom=0,count=0,ambush,locindex,orig;
  byte actorpresent[10]={0},index=0,randomtype,used[100]={0};
  _2Dpoint randloc[100];
+ word randtile[100];
 
 
  map = mapplanes[1];
@@ -5921,6 +5922,7 @@
 		 if ((tile >= 122) && (tile <= 125))
 			 {randloc[totalrandom].x = i;
 			  randloc[totalrandom].y = j;
+			  randtile[totalrandom] = tile;
 			  totalrandom++;
 			  if (totalrandom >= 100)
 				Error("Max random actors (100) exceeded");
@@ -5955,6 +5957,7 @@
 		  ambush = (GameRandomNumber("rand actor",0) < 128);
 		  i = randloc[locindex].x;
 		  j = randloc[locindex].y;
+		  tile = randtile[locindex];
 		  SpawnStand(randomtype,i,j,tile-122,ambush);
 		  used[locindex] = 1;
         PreCacheActor(randomtype,0);
